### PR TITLE
Add more structure to docs for flow

### PR DESCRIPTION
**Pull request by karenzone**
The dissect doc contains good information.  Added structure around it to make it easier to use.

### CODE REVIEW

**Comment by karenzone** [comment]
@dedemorton  Here's a sneak peek.

File: docs/index.asciidoc [hunk 1]
```diff
@@ -20,123 +20,194 @@ include::{include_path}/plugin_header.asciidoc[]
...
+Grok is a better choice when the structure of your text varies from line to line.
+
+You can use both Dissect and Grok for a hybrid use case when a section of the
+line is reliably repeated, but the entire line is not. The Dissect filter can
+deconstruct the section of the line that is  repeated. The Grok filter can process
+the remaining field values with more regex predictability.
+
+===== Terminology
+
+*dissection* - the set of fields and delimiters remaining after passing through the filter.
```
Comment (guyboertje) [inline docs/index.asciidoc]: Not strictly true. In grok it is called a "grok pattern" in dissect I called it a dissection, however in the interests of consistency the perhaps "*dissect pattern* - the set of fields and delimiters defining the structure of a textual format, also known as a dissection".

**Comment by karenzone** [comment]
@guyboertje Thanks for the review comments. I incorporated your changes.
The beats docs used the terms "tokens" and "tokenization."  I'd like to use these terms if possible, in case people are searching on them. Please let me know what you think. Thanks!

**Comment by guyboertje** [comment]
@karenzone 

Go for it!! I think you are far better skilled to craft an educative, elucidatory and enlightening doc than I am.
I use too many big words!

**Comment by karenzone** [comment]
@guyboertje   It's ready for another look when you have time, please.
@dedemorton  Any changes you would like to see?

File: docs/index.asciidoc [hunk 2]
```diff
@@ -20,123 +20,201 @@ include::{include_path}/plugin_header.asciidoc[]
...
-then Grok can be used on the remaining field values with more regex predictability and less overall work to do. +
+The Dissect filter plugin tokenizes incoming strings using defined patterns. 
+It extracts unstructured event data into fields using
+delimiters. This process is called tokenization.
 
-A set of fields and delimiters is called a *dissection*.
+Unlike a regular split operation where one delimiter is applied to the whole
+string, the Dissect operation applies a set of delimiters to a string value. 
 
+The resulting set of fields and delimiters is called a *dissection*.
```
Comment (guyboertje) [inline docs/index.asciidoc]: You can remove this line because of the `Terminology` block, no?

File: docs/index.asciidoc [hunk 3]
```diff
@@ -20,123 +20,201 @@ include::{include_path}/plugin_header.asciidoc[]
...
+
+===== Dissect or Grok? Or both?
+
+Dissect differs from Grok in that it does not use regular expressions and is faster. 
+Dissect works well when data is reliably repeated.
+Grok is a better choice when the structure of your text varies from line to line.
+
+You can use both Dissect and Grok for a hybrid use case when a section of the
+line is reliably repeated, but the entire line is not. The Dissect filter can
+deconstruct the section of the line that is  repeated. The Grok filter can process
```
Comment (guyboertje) [inline docs/index.asciidoc]: `is  repeated`

two spaces :nitpick:

**Pull request closed without merging**